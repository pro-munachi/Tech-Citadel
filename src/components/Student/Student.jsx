import { Box, Flex, Image, Text } from "@chakra-ui/react";
import React from "react";

const Student = ({ data }) => {
  return (
    <Flex mt="60px" alignItems="center" flexWrap="wrap">
      <Box>
        <Image
          src={data?.image}
          alt={data?.name}
          h="230px"
          w="230px"
          maxH="230px"
          maxW="230px"
        />
      </Box>

      <Box ml="20px">
        <Text>Full Name: {data?.name}</Text>
        <Text>Username: {data?.username}</Text>
        <Text>Class: {data?.class}</Text>
      </Box>
    </Flex>
  );
};

export default Student;
