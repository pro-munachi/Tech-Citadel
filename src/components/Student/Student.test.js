import { render } from "@testing-library/react";
import Student from "./Student";
import { BrowserRouter } from "react-router-dom";
import { ChakraProvider, theme } from "@chakra-ui/react";
import store from "../../state/store";
import { Provider } from "react-redux";

const Wrapper = ({ children }) => {
  <ChakraProvider theme={theme}>
    <Provider store={store}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  </ChakraProvider>;
};

test("Test Student", async () => {
  const { asFragment } = render(
    <Wrapper>
      <Student />
    </Wrapper>
  );

  expect(asFragment()).toMatchSnapshot();
});
