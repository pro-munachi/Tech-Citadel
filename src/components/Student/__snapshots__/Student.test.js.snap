// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Student 1`] = `<DocumentFragment />`;
